crates/net/network/src/session/active.rs

```diff
@@ -485,6 +485,9 @@ impl Future for ActiveSession {
             return this.poll_disconnect(cx)
         }
 
+        let smtx = this.to_session_manager.inner().to_owned();
+        let mut to_session_manager_poll_tx = PollSender::new(smtx);
+
```

Comment:
this is an issue because this PollSender is dropped when poll returns, meaning if it's in the process of acquiring a permit it's progress gets lost, 

hmm now this suddenly needs a bit more work, because `to_session_manager` is a MeteredSender, but PollSender requires mpsc::Sender...

so if we want to keep the metrics of MeteredSender we probably need a new helper type `MeteredPollSender` that mimics PollSender

does that make sense?

Reply:
Ah yes it does. Actually originally I changed `MeteredSender` to wrap a `PolledSender` but opted not to because of complexity and it felt out of scope. A separate type makes more sense! I forgot to consider the ownership aspect (too much Go dev lately.. :) ).

I'll get on that.

Reply:
Updated with some assumptions about the above.